### PR TITLE
Reorganize CI workflows

### PR DESCRIPTION
**Pull request by orzechdev**
It updates GitHub workflows.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### Do you want to run more stable tests?
Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation

CONTAINERS=1

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-8074 Introduce testing tools](https://app.clickup.com/t/2549495/SALEOR-8074)

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

File: .github/workflows/test-e2e-on-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+              let tagsInString = removedPullRequestBodyAfterTests[0];
+              tagsInString = tagsInString.split('\n');
+              tagsInString.forEach(line => {
+                if (line.includes('[x]')) tags.push(line.replace(/[0-9]+\. \[x\] /, "@stable+@"))
+              });
+              const tagsToReturn = tags.join(",").toString();
+              return tagsToReturn.replace(/\r/g, '')
+            }catch{
+              return '@critical'
+            }
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply .github/workflows/test-e2e-on-deploy.yml:35]: I'm not sure, for now, I just only moved this code to the other file to clean a bit CI

[the remark this answers is not in the document]
Comment (orzechdev) [reply .github/workflows/test-e2e-on-deploy.yml:35]: I think `removedPullRequestBodyBeforeTests[1].split(...)` may raise an error if `removedPullRequestBodyBeforeTests < 0` then split doesn't exist.

**Review by NyanKiyoshi — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Comment by karola312** [comment]
After separating the deploy job and cypress test job, currently, in the cypress test job we are waiting ~7 minutes for the deployment to finish. It will be additional 7 minutes after every push in all of saleor-dashboard PRs. As far as I know, we are paying for time in these jobs. Before these changes, deployment, and tests, in one job took 18min, after changes we have two jobs that took 9 and 18 minutes. Is that ok for you @patrys , @michalina-graczyk

**Comment by patrys** [comment]
@karola312 We could also use https://github.com/marketplace/actions/wait-for-deployment to wait for a deployment to finish.

**Comment by karola312** [comment]
But it still will be a job, which is doing nothing else than waiting for other job to finish for few minutes. Maybe after deployment, we should send a request (/repos/saleor/saleor-dashboard/dispatches), which will trigger the cypress job?

**Comment by orzechdev** [comment]
@patrys It is possible, but we will need a little bit tricky workaround, similar to https://github.com/orgs/community/discussions/25220#discussioncomment-3246922 or use https://github.com/marketplace/actions/get-current-pull-request

**Comment by patrys** [comment]
I think it still beats paying for several minutes of a worker spinning idly.

**Pull request closed without merging**